src/executables/util_cs.c

```diff
@@ -4544,18 +4545,126 @@ flashback (UTIL_FUNCTION_ARG * arg)
 int
 memmon (UTIL_FUNCTION_ARG * arg)
 {
+#if defined(CS_MODE)
   UTIL_ARG_MAP *arg_map = arg->arg_map;
-  char *module;
-  bool transaction, show_all;
+  char er_msg_file[PATH_MAX];
+  const char *database_name;
+  char *module, *stop;
+  bool transaction, show_all, need_shutdown;
   int tran_count;
+  int module_index = MMON_MODULE_LAST;
 
   module = utility_get_option_string_value (arg_map, MEMMON_MODULE_S, 0);
   transaction = utility_get_option_bool_value (arg_map, MEMMON_TRANSACTION_S);
   tran_count = utility_get_option_int_value (arg_map, MEMMON_TRAN_COUNT_S);
   show_all = utility_get_option_bool_value (arg_map, MEMMON_SHOW_ALL_S);
 
+  database_name = utility_get_option_string_value (arg_map, OPTION_STRING_TABLE, 0);
+  if (database_name == NULL)
+    {
+      goto print_memmon_usage;
+    }
+
+  /* error check phase */
+  if (check_database_name (database_name))
+    {
+      goto error_exit;
+    }
+
+  if ((!transaction) && tran_count)
+    {
+      PRINT_AND_LOG_ERR_MSG (msgcat_message
+			     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_INVALID_TRAN_COUNT_OPTION));
+      goto error_exit;
+    }
+
+  if (tran_count < 0)
+    {
+      PRINT_AND_LOG_ERR_MSG (msgcat_message
+			     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_NEGATIVE_TRAN_COUNT_NUMBER));
+      goto error_exit;
+    }
+
+  if (show_all && (transaction || module))
+    {
+      PRINT_AND_LOG_ERR_MSG (msgcat_message
+			     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_EXCLUSIVE_OPTION));
+      goto error_exit;
+    }
+
+  if (module)
+    {
+      module_index = strtol (module, &stop, 10);
+      if (module_index < 0 || module_index > MMON_MODULE_LAST)
+	{
+	  PRINT_AND_LOG_ERR_MSG (msgcat_message
+				 (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_NO_MATCHING_MODULE), module);
+	  goto error_exit;
+	}
+      else if (module[0] > 'a' && module[0] < 'z')
+	{
+	  /* module option with module name */
+	  // TODO: convert module name to module index
+	  // ex) module_index = memmon_convert_module_name_to_index(module);
+	  if (module_index == -1)
+	    {
+	      PRINT_AND_LOG_ERR_MSG (msgcat_message
+				     (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_NO_MATCHING_MODULE),
+				     module);
+	      goto error_exit;
+	    }
+	}
+    }
```

Comment:
Is this working correctly? What is the value of the module_index if -m all is set?

Reply:
Do you mean the code which checks the module_index/module_name will be changed correctlly?